src/main/java/net/minestom/server/listener/SpectateListener.java

```diff
@@ -16,7 +16,12 @@ public static void listener(ClientSpectatePacket packet, Player player) {
         if (target == null || target == player)
             return;
 
-        // TODO check if 'target' is in a different instance
+        // Check if the target is in a different instance
+        if (target.getInstance() != player.getInstance()) {
+            //noinspection ConstantConditions
+            player.setInstance(target.getInstance()).thenRun(() -> player.spectate(target));
```

Comment:
Doesn't this let the client switch instances if it knows the UUID of a player in that instance and is in spectator mode? Or are they shown in the middle click menu?

Reply:
That's a good point, they would be able to do that. I'm not sure if it should be removed though... E.g. the vanilla server allows this.

Maybe adding an event for users to decide is a good solution.

Reply:
Yes it does, however the way instances are handled in for example the ArgumentEntity makes me believe that instances are not like vanilla dimensions but rather like vanilla servers with only one dimension

Reply:
Yep I think I'll have that behaviour by default with the ability to change it via an event.